index.js

```diff
@@ -25,61 +25,61 @@ instance.prototype.updateConfig = function(config) {
 	self.init_tcp();
 };
 
+instance.prototype.init = function() {
+	var self = this;
+
+	debug = self.debug;
+	log = self.log;
+
+	self.init_tcp();
+};
+
 instance.prototype.incomingData = function(data) {
 	var self = this;
 	debug(data);
 
 	// Match part of the copyright response from unit when a connection is made.
-	// Send Info request which should reply with Matrix setup, eg: "V8X4 A8X4"
 	if (self.login === false && data.match("Extron Electronics")) {
 		self.status(self.STATUS_WARNING,'Logging in');
-		self.socket.write("I"+ "\n");
+		self.socket.write("I"+ "\n"); // Matrix information request
 	}
 
 	if (self.login === false && data.match("Password:")) {
 		self.status(self.STATUS_WARNING,'Logging in');
-		self.socket.write(""+ "\n");
+		self.socket.write("\r" +self.config.password+ "\r"); // Enter Password Set
```

Comment:
A bit weird that the password would be entered with "carriage return" \r, when all other commands are using only "line feed" \n ? Is this per spec? And why the carriage return before the password? Is this tested?

Reply:
I did tests on my SMX with hercules as it didn't seem to work in companion. So I could see responses \r made things work and seemed to be needed.  \r beford password guarantees it's waiting for the password with a clean line, it wasn't accepting it without it. This is same tested working code I added to the SMP module as it was requested, thought I'd have a go :) What's the worst that can happen! (DOGHOUSE)

Reply:
Also setting verbose mode requires \r and needs to be the first command sent which is what gave me the idea to try it!

Reply:
Ok :)